signer/src/stacks/wallet.rs

```diff
@@ -397,14 +404,33 @@ pub fn get_full_tx_size<T>(payload: &T, wallet: &SignerWallet) -> Result<u64, Er
 where
     T: AsTxPayload,
 {
-    use rand::SeedableRng as _;
     use stacks_common::codec::StacksMessageCodec as _;
-    let mut rng = rand::rngs::StdRng::seed_from_u64(1);
-    // We need some private keys, that we control, so let's create them.
     let num_signers = wallet.public_keys().len();
-    let private_keys = std::iter::repeat_with(|| PrivateKey::new(&mut rng))
+
+    // We need some dummy private keys to sign the transaction. We check if we've
+    // already generated enough dummy keys, and if not, we generate more. This
+    // is to ensure that always have enough dummy keys even if the signing set
+    // grows.
+    let mut dummy_keys = DUMMY_PRIVATE_KEYS
+        .lock()
+        .expect("BUG! DUMMY_PRIVATE_KEYS lock poisoned");
```

Comment:
Ah I actually had a LazyCell first but wasn't sure if we should have a cap for key count, but sounds like we have it capped already. Otherwise this impl was just due to my experience of generating private keys to generally be computationally expensive, but it sounds like ecsda256 keys are cheap.

I like the LazyCell solution the best so I'll toss that in :+1: